src/Features/Tas/TasTools.cpp

```diff
@@ -0,0 +1,148 @@
+#include "TasTools.hpp"
+
+#include <cmath>
+
+#include "Features/Session.hpp"
+#include "Modules/Client.hpp"
+#include "Modules/Engine.hpp"
+#include "Modules/Server.hpp"
+
+#include "Utils/Math.hpp"
+#include "Utils/SDK.hpp"
+#include <Offsets.hpp>
+
+TasTools* tasTools;
+
+TasTools::TasTools()
+{
+    this->hasLoaded = true;
+    this->m_previous_speed = Vector{ 0, 0, 0 };
+    this->m_last_tick = (session->isInSession) ? engine->GetSessionTick() : 0;
+    this->m_acceleration = Vector{ 0, 0, 0 };
+
+    strcpy(this->m_offset_name, "m_InAirState");
+    strcpy(this->m_class_name, "CPortal_Player");
+}
+void TasTools::AimAtPoint(float x, float y, float z)
+{
+    Vector target = { y, x, z };
+    Vector campos = client->GetAbsOrigin() + client->GetViewOffset();
+    campos = { campos.y, campos.x, campos.z };
+
+    // Axis in the game, need to know it to fix up:
+    //              : G - D ; Av - Ar ; H - B
+    // Rotation Axis:   x        z        y
+    // Translation  :   y        x        z
+
+    float xdis = target.x - campos.x;
+    float ydis = target.z - campos.z;
+    float zdis = target.y - campos.y;
+    float xzdis = sqrtf(xdis * xdis + zdis * zdis);
+
+    QAngle angles = { RAD2DEG(-atan2f(ydis, xzdis)), RAD2DEG(-(atan2f(-xdis, zdis))), 0 };
+
+    engine->SetAngles(angles);
+}
+
+// Commands
+
+CON_COMMAND(sar_tas_aim_at_point, "sar_tas_aim_at_point <x> <y> <z> : Aim at the point {x, y, z} specified.\n")
+{
+    if (!sv_cheats.GetBool()) {
+        return console->Print("Cannot use sar_tas_aim_at_point without sv_cheats set to 1.\n");
+    }
+
+    if (args.ArgC() != 4) {
+        return console->Print("sar_tas_aim_at_point <x> <y> <z> : Aim at the point {x, y, z} specified.\n");
+    }
+
+    tasTools->AimAtPoint(static_cast<float>(atof(args[1])), static_cast<float>(atof(args[2])), static_cast<float>(atof(args[3])));
+}
+
+
+int TasTools::GetOffset()
+{
+    client->GetOffset(this->m_class_name, this->m_offset_name, Offsets::anyOffset);
+    auto player = client->GetPlayer();
```

Comment:
Right now it's kinda pointless to set `m_class_name` to anything else than `CBasePlayer` or `CPortal_Player` so you could check that in the command where you set the offset